engine/access/rest/websockets/data_providers/data_provider.go

```diff
@@ -0,0 +1,11 @@
+package data_providers
+
+// The DataProvider is the interface abstracts of the actual data provider used by the WebSocketCollector.
+type DataProvider interface {
+	BaseDataProvider
+
+	// Run starts processing the subscription and handles responses.
+	//
+	// No errors are expected during normal operations.
+	Run() error
```

Comment:
@peterargue, the idea behind separating the creation of the data provider and the `Run()` method is to provide better control over the subscription lifecycle. By doing this, we can immediately send a message to the client confirming that the subscription has been successfully created or failed. The actual subscription and data streaming process will only start after calling `Run()`, which ensures that we can handle any setup or preparation steps first. Since `Run()` starts processing the subscription, we do not need the `ctx` at this point.

Reply:
can you add a comment with this context in `BaseDataProvider` or wherever you think makes it most clear. I think future readers will wonder this as well.

Reply:
[437af4f](https://github.com/onflow/flow-go/pull/6636/commits/437af4f6cdb52aaddc8b07cb80943b93c88dceed)